tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * @group editor
+ *
+ * @covers ::_enable_content_editor_for_navigation_post_type
+ */
+class Tests_Editor_EnableContentEditorForNavigationPostType extends WP_UnitTestCase {
+
+	const NAVIGATION_POST_TYPE     = 'wp_navigation';
+	const NON_NAVIGATION_POST_TYPE = 'wp_non_navigation';
+
+	public function tear_down() {
+		$this->enable_editor_support();
+		parent::tear_down();
+	}
+
+	/**
+	 * @ticket 56266
+	 *
+	 * @dataProvider data_test_it_correctly_handles_different_post_types
+	 */
+	public function test_it_correctly_handles_different_post_types( $post_type, $enable_editor_support, $expected, $message ) {
+		$post = $this->create_post( $post_type );
+		if ( $enable_editor_support ) {
+			$this->enable_editor_support();
+			$this->assertTrue( $this->supports_block_editor(), 'Editor support must be enabled before running the test.' );
+		} else {
+			$this->disable_editor_support();
+			$this->assertFalse( $this->supports_block_editor(), 'Editor support must be disabled before running the test.' );
+		}
+
+		_enable_content_editor_for_navigation_post_type( $post );
+
+		$this->assertSame( $expected, $this->supports_block_editor(), $message );
+	}
+
+	/**
+	 * @ticket 56266
```

Comment:
```suggestion
	 * Data provider.
	 *
	 * @return array
```

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/31788a26830266e44de78e6decb864d6a228e2d0